src/garage/tf/models/gaussian_gru_model2.py

```diff
@@ -0,0 +1,236 @@
+            step_input (tf.Tensor): Single timestep observation input.
+            hidden_input (tf.Tensor): Hidden state for step.
+
+        Returns:
+            tf.Tensor: Entire time-series means.
+            tf.Tensor: Step means.
+            tf.Tensor: Entire time-series log std.
+            tf.Tensor: Step log std.
+            tf.Tensor: Step hidden state.
+            tf.Tensor: Initial hidden state.
+            tfp.distributions.MultivariateNormalDiag: Distribution.
+
+        """
+        action_dim = self._output_dim
+
+        with tf.compat.v1.variable_scope('dist_params'):
+            if self._std_share_network:
+                # mean and std networks share an MLP
+                (outputs, step_outputs, step_hidden, hidden_init_var) = gru(
+                    name='mean_std_network',
+                    gru_cell=self._mean_std_gru_cell,
+                    all_input_var=state_input,
+                    step_input_var=step_input,
+                    step_hidden_var=hidden_input,
+                    hidden_state_init=self._hidden_state_init,
+                    hidden_state_init_trainable=self.
+                    _hidden_state_init_trainable,
+                    output_nonlinearity_layer=self.
+                    _mean_std_output_nonlinearity_layer)
+                with tf.compat.v1.variable_scope('mean_network'):
+                    mean_var = outputs[..., :action_dim]
+                    step_mean_var = step_outputs[..., :action_dim]
+                with tf.compat.v1.variable_scope('log_std_network'):
+                    log_std_var = outputs[..., action_dim:]
+                    step_log_std_var = step_outputs[..., action_dim:]
+
+            else:
+                # separate MLPs for mean and std networks
+                # mean network
+                (mean_var, step_mean_var, step_hidden, hidden_init_var) = gru(
+                    name='mean_network',
+                    gru_cell=self._mean_gru_cell,
+                    all_input_var=state_input,
+                    step_input_var=step_input,
+                    step_hidden_var=hidden_input,
+                    hidden_state_init=self._hidden_state_init,
+                    hidden_state_init_trainable=self.
+                    _hidden_state_init_trainable,
+                    output_nonlinearity_layer=self.
+                    _mean_output_nonlinearity_layer)
+                log_std_var, step_log_std_var = recurrent_parameter(
+                    input_var=state_input,
+                    step_input_var=step_input,
+                    length=action_dim,
+                    initializer=tf.constant_initializer(self._init_std_param),
+                    trainable=self._learn_std,
+                    name='log_std_param')
+
+        dist = tfp.distributions.MultivariateNormalDiag(
+            loc=mean_var, scale_diag=tf.exp(log_std_var))
+
+        return (mean_var, step_mean_var, log_std_var, step_log_std_var,
```

Comment:
why return `mean_var` and `log_std_var` when i can get them from `dist`?

Reply:
fixed, now it doesn't return them